tests/fixtures/entities.py

```diff
@@ -37,3 +39,73 @@ def instance_attrs(instance, i):
 
     return functools.partial(
         factory, Model=Entity, instance_attrs=instance_attrs)
+
+
+@pytest.fixture
+def entity_models(translation0, locale1):
+    entity0 = translation0.entity
+    locale0 = translation0.locale
+    project0 = entity0.resource.project
+
+    locale0.cldr_plurals = "0,1"
+    locale0.save()
+    translation0.plural_form = 0
+    translation0.save()
+    resourceX = Resource.objects.create(
+        project=project0, path="resourceX.po")
+    entity0.string = "Entity zero"
+    entity0.key = entity0.string
+    entity0.string_plural = "Plural %s" % entity0.string
+    entity0.save()
+    entityX = Entity.objects.create(
+        resource=resourceX,
+        string="entityX",
+        key='Key%sentityX' % KEY_SEPARATOR)
+    translation0pl = Translation.objects.create(
+        entity=entity0,
+        locale=locale0,
+        plural_form=1,
+        string="Plural %s" % translation0.string)
+    translationX = Translation.objects.create(
+        entity=entityX,
+        locale=locale0,
+        string="Translation %s" % entityX.string)
+    subpageX = Subpage.objects.create(
+        project=project0, name="Subpage")
+    subpageX.resources.add(entity0.resource)
+    return translation0, translation0pl, translationX, subpageX
+
+
+@pytest.fixture
+def entity_search(entity_factory, translation_factory, resourceX, localeX):
```

Comment:
Same here, I'd love to see a docstring here. Actually my first reaction that this is a bit too smart for our good, but maybe it's because I don't have much experience with fixtures. I'm surprised to have a fixture that returns a function.

Reply:
I see no problem at all with a fixture returning a function - this is functional programming after all